### ISSUE TITLE
Consider whether lists can be initialized to empty instead of null

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/csla#4266]
> > Another question: Any good reason why this lists can't just be initialized and hence empty if not neede and/or used? Makes the nullable stuff a lot easier and even more intuitive to use. Currently accessing `DirtyProperties` in a rule would return `null` instead of an empty list...
> > https://github.com/MarimerLLC/csla/blob/aa14c3f86e3052d73cb06785b6fb6c381e6eb9f2/Source/Csla/Rules/RuleContext.cs#L84-L98
> > 
> > https://github.com/MarimerLLC/csla/blob/aa14c3f86e3052d73cb06785b6fb6c381e6eb9f2/Source/Csla/Rules/RuleContext.cs#L66-L82
> 
> CSLA often doesn't initialize lists of types that might be serialized, because it is _much_ smaller to serialize `null` than it is to serialize an empty list. This is because the serializer doesn't need to flow the list's type information for `null`, but it does need to flow the type information for an empty list so the empty list can be deserialized on the other end of the process.

For `MobileFormatter`, yes, you need to serialize type information for a collection type so you know what to create on the other side. But that isn't a **strict** requirement for CSLA serialization. For example, my source generator serializer doesn't require this, because it knows at compile-time what the underlying type is for the field, and can `new` it up how it sees fit. For collections like `List<T>`, I don't special-case that one, but it's trivial to add a custom implementation for that in my generator through configuration.

Also, I do like empty collections over `null` in general overall. Then I don't need to wonder, "is this a `null` list or not?" everywhere in my code. If the performance implications of a `null` list over an empty one is significant during serialization, then....maybe keep them null, but if it's essentially negligible, then I'd **much rather** see an empty list.

_Originally posted by @JasonBock in https://github.com/MarimerLLC/csla/issues/1233#issuecomment-2429212811_

**Comment by StefanOssendorf** [comment]
I agree. I find it really weird to have a list `null` instead of just empty. What exactly is the difference in use?
If we want to keep/improve (de)serialization speed, we can special-case empty lists etc. so `null` is transported and we just re-initialize it on the other side.

**Comment by rockfordlhotka** [comment]
Null and empty aren't the same thing, so I don’t think we can translate arbitrarily. I kind of suspect that would affect lazy loading, though that'll require some exploration.

It isn't the perfect of serialization that's in question, it is the size of the payload. And that may be incidental, so metrics need to be established.
